### PR TITLE
fix(ui): Overlapping buttons at narrow screen widths

### LINKED ISSUE #4906: Filter buttons overlap View buttons when window is narrow
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

The application details view has a bunch of left aligned buttons and a bunch of right aligned buttons. There isn't enough space reserved to ensure they don't overlap.

**To Reproduce**

1. Visit https://cd.apps.argoproj.io/applications/argo-cd
1. Adjust window size / font zoom until problem appears

**Expected behavior**

Buttons shouldn't overlap

**Screenshots**

filter button shines through delete -- very dangerous and in from log-in sticks out

**Version**

```shell
{
    "Version": "v1.8.0+a8a451f",
    "BuildDate": "2020-11-24T18:26:08Z",
    "GitCommit": "a8a451f8113f9d5085b3c190f5c58c86981854b2",
    "GitTreeState": "clean",
    "GoVersion": "go1.14.12",
    "Compiler": "gc",
    "Platform": "linux/amd64",
    "KsonnetVersion": "v0.13.1",
    "KustomizeVersion": "v3.8.1 2020-07-16T00:58:46Z",
    "HelmVersion": "v3.4.0+g7090a89",
    "KubectlVersion": "v1.17.8",
    "JsonnetVersion": "v0.16.0"
}
```

**Logs**

```
Paste any relevant application logs here.
```

### PR DESCRIPTION
**Pull request by rbreeze**
Also addresses other various visual artifacts at narrow browser window widths. Demo of update below: 
Jan-15-2021 16-22-40

Fix #4906

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [x] My build is green ([troubleshooting builds](https://argoproj.github.io/argo-cd/developer-guide/ci/)).

### CODE REVIEW

**Comment by jsoref** [comment]
If you could avoid the scrolling in favor of the compacting (for the top buttons) or wrapping (for the lower three boxes), that'd be better.

But this is still a huge improvement.

**Comment by rbreeze** [comment]
Fixed to compact buttons rather than have scrolling. I left the lower 3 boxes with a scroll because to wrap them would mean pushing all the information below that status bar off the screen, which I don't like.

**Comment by jsoref** [comment]
Eventually one should imagine viewing this on a phone. On phones those three boxes should end up stacked vertically, and the content below them pushed down further (side scrolling is really bad on mobile). But, for now, this is an incremental improvement.

One thing you could do to address the offscreen problem is:

<details>
<summary> 🔵  Progressing</summary>
> The lines that you
> would normally see
> in progressing
</details>

<details>
<summary> ✅  Synced </summary>
> The lines that you
> would normally see
> in synced
</details>

<details>
<summary> ❌   Sync Error </summary>
> The lines that you
> would normally see
> in Sync Error
</details>

---

_Content Area_

**Review by alexmt — CHANGES_REQUESTED** [review]
Tested locally and noticed strange horizontal bar between button rows:


image

I think because of that row `Group By` label is semi-hidden. Can you please fix it?

**Comment by rbreeze** [comment]
What browser are you using? I can't seem to reproduce on Chrome, Firefox or Safari

**Comment by alexmt** [comment]
I use chrome, tried again. Here is video.

https://user-images.githubusercontent.com/426437/106218111-8aaeec00-618b-11eb-8dd6-4a1c7af3ffc7.mov

hopefully it can help reproducing it

**Comment by jsoref** [comment]
That horizontal bar looks like a scrollbar gutter.

**Comment by jsoref** [comment]
@alexmt: can you try incognito? (morbidly curious)

@rbreeze: maybe change the `overflow-x` to something other than `scroll`?

**Comment by rbreeze** [comment]
hmm, I just tried in Chrome again and still can't see it: 

https://user-images.githubusercontent.com/6250584/106317123-ab765080-6222-11eb-931f-30d6430def9d.mov

Could you try maybe clearing the cache to see if there's potential old CSS causing it?

**Comment by rbreeze** [comment]
Ah, I think I found the problem. Do you have this setting set on your Mac? 

Screen Shot 2021-01-29 at 11 14 50 AM

Turning this on reproduced it for me. I'll find a fix

**Comment by jsoref** [comment]
Yeah, that's what I was expecting https://github.com/argoproj/argo-cd/pull/5259#issuecomment-769520123 👍

File: ui/src/app/applications/components/application-status-panel/application-status-panel.scss [hunk 1]
```diff
@@ -74,9 +88,42 @@
...
     }
-    &__item-name, &__item-value {
+    &__item-name,
+    &__item-value {
         overflow: hidden;
         text-overflow: ellipsis;
         white-space: nowrap;
     }
+
+    & .info-popup-title {
```
Comment (alexmt) [inline ui/src/app/applications/components/application-status-panel/application-status-panel.scss:98]: CSS styles started from this line were deleted in this commit: https://github.com/argoproj/argo-cd/commit/7165b45aeaa15044cec514e7de3d300d448b2df9#diff-4b88f846774c449afe371a91f0a5184f764cc1cc98a947003281c74407dc4074L82

Please delete obsolte styles

**Review by alexmt — CHANGES_REQUESTED** [review]
Looks awesome! one last nitpick

**Review by alexmt — APPROVED** [review]
LGTM

**Pull request merged**